src/commands/Command.ts

```diff
@@ -0,0 +1,396 @@
+                let allCommands = [...commands, this];
+
+                let prefixes = [];
+                let width = 0;
+                let mainPrefix = getMainPrefix();
+
+                // Compute width to display the help properly.
+                for (let command of allCommands) {
+                    let prefix = `${mainPrefix} ${command.command} ${command.helpArgs} `;
+                    width = Math.max(width, prefix.length);
+                    prefixes.push(prefix);
+                }
+
+                // Now build actual help message.
+                let lines = [];
+                for (let i = 0; i < prefixes.length; ++i) {
+                    let prefix = prefixes[i].padEnd(width);
+                    let line = `${prefix} - ${allCommands[i].helpDescription}`;
+                    lines.push(line);
+                }
+
+                let message = lines.join("\n");
+                const html = `<b>Mjolnir help:</b><br><pre><code>${htmlEscape(message)}</code></pre>`;
+                const text = `Mjolnir help:\n${message}`;
+                const reply = RichReply.createFor(roomID, event, text, html);
+                reply["msgtype"] = "m.notice";
+                await mjolnir.client.sendMessage(roomID, reply);    
+            }
+        }
+        this.helpCommand = new HelpCommand();
+    }
+
+    public async init(prefixes: string[]) {
+        // Prepare prefixes.
+        this.prefixes.length = 0;
+        for (let prefix of prefixes) {
+            let lowercase = prefix.trim().toLowerCase();
+            if (!lowercase.startsWith("!")) {
+                // Note: This means that if the prefix is `!mjolnir`, we will also
+                // respond to `!mjolniren` or any other suffixed variant.
+                this.prefixes.push(`!${lowercase}`);
+            }
+            if (!lowercase.endsWith(":")) {
+                this.prefixes.push(`${lowercase}:`);
+            }
+            this.prefixes.push(`${lowercase} `);
+        }
+        
+        this.mjolnir.client.on("room.message", this.onMessageCallback);
+    }
+
+    public async dispose() {
+        this.mjolnir.client.removeListener("room.message", this.onMessageCallback);
+    }
+
+    /**
+     * Handle messages in any room to which we belong.
+     *
+     * @param roomId The room in which the message is received.
+     * @param event An untrusted event.
+     */
+    private async handleMessage(roomId: string, event: any) {
+        try {
+            if (roomId != this.managementRoomId) {
+                // Security-critical: We only ever accept commands from our management room.
+                return;
+            }
+            const content = event['content'];
+            if (!content || content['msgtype'] !== "m.text" || content['body']) {
+                return;
+            }
+
+            const body = content['body'];
+            const lowercaseBody = body.toLowerCase();
```

Comment:
Are we sure we want to do that? Did we do it before?

Reply:
Yes, that's for backwards compatibility.